src/mainloop.rkt

```diff
@@ -423,16 +423,25 @@
     (simplify-batch
       (make-egg-query (map (compose program-body alt-program) joined-alts) (*fp-safe-simplify-rules*)) #t))
   (define cleaned-alts
```

Comment:
I feel like some of this stuff should have been deleted?

Reply:
Let me take a look